src/components/graph-view.js

```diff
@@ -1586,13 +1607,50 @@ class GraphView extends React.Component<IGraphViewProps, IGraphViewState> {
     );
   }
 
-  handlePanStart(event: any) {
+  handlePanWheel = (event: any) => {
+    // don't allow pan / wheel to affect document
+    event.stopPropagation();
+    event.stopImmediatePropagation();
+    event.preventDefault();
+
+    if (this.wheelState.requestId != null) {
+      cancelAnimationFrame(this.wheelState.requestId);
+    }
+
+    let requestId = null;
+
+    if (!this.wheelState.zooming) {
+      requestId = requestAnimationFrame(() => {
+        const { viewTransform } = this.state;
+        const offX = this.wheelState.deltaX + viewTransform.x;
+        const offY = this.wheelState.deltaY + viewTransform.y;
+
+        this.wheelState = {
+          zooming: true,
+          requestId: null,
+          deltaX: 0,
+          deltaY: 0,
+        };
+
+        this.setZoom(viewTransform.k || 1, offX, offY, 0);
+      });
+    }
+
+    this.wheelState = {
+      requestId,
+      zooming: this.wheelState.zooming,
```

Comment:
when do make this.wheelState.zooming to true, am I missing something

Reply:
We set `this.wheelState = { zooming: true ... }` above. We set it to false in `handleZoomEnd`.